### PR TITLE
Fix tests for gslib miniapps

### PR DESCRIPTION
**Pull request by kmittal2**
This PR address #2940.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge|
 | --- | --- | --- | --- | --- | --- | --- |
| [#2945](https://github.com/mfem/mfem/pull/2945) | @kmittal2 | @tzanio | @v-dobrev + @psocratis | 3/30/22 | 5/10/22 | 5/13/22 | |

PR Checklist
    
- [x] Code builds.
- [x] Code passes `make style`.

### CODE REVIEW

**Comment by termi-official** [comment]
Quick question: This PR is primarily about adding tests to the gslib miniapps and not about fixing lingering bugs, right?

**Comment by termi-official** [comment]
Quick question: This PR is primarily about adding tests to the gslib miniapps and not about fixing lingering bugs, right?

**Comment by kmittal2** [comment]
> Quick question: This PR is primarily about adding tests to the gslib miniapps and not about fixing lingering bugs, right?

Hi @termi-official, yes the purpose of this PR is to add tests to the gslib miniapps. Can you please elaborate on what lingering bugs you are talking about?

**Comment by termi-official** [comment]
Thanks for the quick response @kmittal2 !

I do not have a minimum working example at hand - Basically I run into issues that the "field-interp" miniapp messes up the transfer of L2 and H1 grid functions, at least on some of my meshes. I have it on my todo list to put together a detailed bug report, but I cannot find the time ATM to track down what goes wrong. I will try to put something together about mid may.

Regarding the changes: They look good to me.

**Comment by kmittal2** [comment]
Sounds good @termi-official. Please open an issue as soon as you get a chance.

File: miniapps/gslib/CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +43,14 @@ if (MFEM_USE_GSLIB)
   add_mfem_miniapp(field-interp
     MAIN field-interp.cpp
     LIBRARIES mfem)
+  add_dependencies(field-diff copy_miniapps_gslib_data)
+
+  if (MFEM_ENABLE_TESTING)
+    foreach (test "schwarz_ex1" "field-diff" "findpts" "field-interp")
+      add_test(NAME ${test}}
```
Comment (v-dobrev) [inline miniapps/gslib/CMakeLists.txt:50]: `}}` -> `}`

Comment (kmittal2) [reply miniapps/gslib/CMakeLists.txt]: Good catch!

**Review by psocratis — APPROVED** [review]

**Comment by tzanio** [comment]
@v-dobrev and @termi-official, can I get another review of this PR?

**Comment by psocratis** [comment]
@v-dobrev, I double checked the gslib tests. All look ok now.

`````
GSLIB miniapp [ mpirun -np 4 pfindpts ... ]: OK  (1.04s 18464kB)
GSLIB miniapp [ mpirun -np 4 schwarz_ex1p ... ]: OK  (1.23s 18508kB)
GSLIB miniapp [ findpts ... ]: OK  (0.96s 11056kB)
GSLIB miniapp [ field-diff ... ]: OK  (0.93s 15420kB)
GSLIB miniapp [ field-interp ... ]: OK  (1.00s 10880kB)
GSLIB miniapp [ schwarz_ex1 ... ]: OK  (1.01s 11812kB)
`````

**Comment by termi-official** [comment]
Tests are broken on my side due to a bug in Catch2 (see https://github.com/catchorg/Catch2/issues/2178). 

I was able to resolve the issue locally by replacing the single header in the test folder with the latest single header version (i.e. https://raw.githubusercontent.com/catchorg/Catch2/v2.x/single_include/catch2/catch.hpp). Could you update the Catch2 dependency to a newer version?

**Comment by tzanio** [comment]
☝️ ping: @pazner

**Review by v-dobrev — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**